pkg/controller/case_test.go

```diff
@@ -0,0 +1,583 @@
+}
+
+func (ct *ControllerTest) TearDown() {
+	close(ct.stopCh)
+}
+
+func (ct *ControllerTest) AsyncForInstances() {
+	ct.fakeOSBClient.ProvisionReaction.(*fakeosb.ProvisionReaction).Response.Async = true
+	ct.fakeOSBClient.UpdateInstanceReaction.(*fakeosb.UpdateInstanceReaction).Response.Async = true
+	ct.fakeOSBClient.DeprovisionReaction.(*fakeosb.DeprovisionReaction).Response.Async = true
+}
+
+func (ct *ControllerTest) AsyncForBindings() {
+	ct.fakeOSBClient.BindReaction.(*fakeosb.BindReaction).Response.Async = true
+	ct.fakeOSBClient.UnbindReaction.(*fakeosb.UnbindReaction).Response.Async = true
+}
+
+func (ct *ControllerTest) AssertOSBBasicAuth(t *testing.T, username, password string) {
+	require.NotNil(t, ct.osbClientCfg, "OSB Client was not created, wait for broker is ready")
+	assert.Equal(t, ct.osbClientCfg.AuthConfig.BasicAuthConfig, &v2.BasicAuthConfig{
+		Username: username,
+		Password: password,
+	})
+}
+
+func (ct *ControllerTest) SetFirstOSBPollLastOperationReactionsInProgress(numberOfInProgressResponses int) {
+	numberOfPolls := 0
+	ct.fakeOSBClient.PollLastOperationReaction = fakeosb.DynamicPollLastOperationReaction(
+		func(_ *osb.LastOperationRequest) (*osb.LastOperationResponse, error) {
+			numberOfPolls++
+			state := osb.StateInProgress
+			if numberOfPolls > numberOfInProgressResponses {
+				state = osb.StateSucceeded
+			}
+			return &osb.LastOperationResponse{State: state}, nil
+		})
+}
+
+func (ct *ControllerTest) SetFirstOSBProvisionReactionsHTTPError(numberOfErrorResponses int) {
+	numberOfPolls := 0
+	ct.fakeOSBClient.ProvisionReaction = fakeosb.DynamicProvisionReaction(
+		func(_ *osb.ProvisionRequest) (*osb.ProvisionResponse, error) {
+			numberOfPolls++
+			if numberOfPolls > numberOfErrorResponses {
+				return &osb.ProvisionResponse{}, nil
+			}
+			return nil, osb.HTTPStatusCodeError{
+				StatusCode: http.StatusUnauthorized,
+			}
+		})
+}
+
+func (ct *ControllerTest) SetOSBBindReactionWithHTTPError(code int) {
+	ct.fakeOSBClient.BindReaction = &fakeosb.BindReaction{
+		Error: osb.HTTPStatusCodeError{
+			StatusCode: code,
+		},
+	}
+}
+
+func (ct *ControllerTest) spyOSBClientFunc(target v2.CreateFunc) v2.CreateFunc {
+	return func(osbCfg *v2.ClientConfiguration) (v2.Client, error) {
+		ct.osbClientCfg = osbCfg
+		return target(osbCfg)
+	}
+}
+
+func (ct *ControllerTest) createClusterServiceBroker() *v1beta1.ClusterServiceBroker {
+	return &v1beta1.ClusterServiceBroker{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testClusterServiceBrokerName,
+		},
+		Spec: v1beta1.ClusterServiceBrokerSpec{
+			CommonServiceBrokerSpec: v1beta1.CommonServiceBrokerSpec{
+				URL:            "https://broker.example.com",
+				RelistBehavior: v1beta1.ServiceBrokerRelistBehaviorDuration,
+				RelistDuration: &metav1.Duration{Duration: 15 * time.Minute},
+			},
+		},
+	}
+}
+
+func (ct *ControllerTest) CreateSimpleClusterServiceBroker() {
+	ct.scInterface.ClusterServiceBrokers().Create(ct.createClusterServiceBroker())
```

Comment:
why you are not returning the error? it can be important in test to assert that.  I know that is a fake client but still, it can produce an error (e.g. when u add ProxyReactor)

This comment applies for all `Create*` functions

Reply:
ok, I've added error to methods Create*